### ISSUE TITLE
`java.lang.NoClassDefFoundError: nu/validator/vendor/thaiopensource/validate/Schema`

### ISSUE THREAD

**Issue by boris-petrov** [validator/validator#2008]
### Summary

Ever since #1504 was fixed, I've been getting this error.

I'm using the Maven release.

Which is indeed true - there is no such file in the JAR file. I see `com.thaiopensource.validate.Schema.class`. So perhaps something hasn't been done right with the shading part, at least in the Maven release.

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
n/a
```

### Checker error messages

```shell
n/a
```

### Steps to reproduce

n/a

### Expected behavior

n/a

### Actual behavior

n/a

**Comment by sideshowbarker** [comment]
Thanks much for catching this and reporting it. I’ve fixed in the sources, and will a new version to Maven Central tomorrow.

Changes:

- Use shaded jing-trang jar in Maven artifacts (build/build.xml)
- Update service providers to use shaded package names
- Add missing commons-text dependency to Maven POM
- Create Maven integration test suite (tests/maven-integration/)
- Add maven-test task to both Python (checker.py) and Ant (build.xml)
- Integrate maven-test into CI workflow (pre-release verification)

The Maven artifact now correctly includes shaded jing-trang classes at nu.validator.vendor.thaiopensource.* instead of unshaded classes at com.thaiopensource.* — preventing a runtime ClassNotFoundException

The test suite verifies:

- Shaded classes are present
- Unshaded classes ain’t present
- EmbeddedValidator works correctly
- Actually checking an HTML file with it works as expected

**Comment by boris-petrov** [comment]
Thank you for the quick and comprehensive fix! I'll be sure to try it out tomorrow and let you know if there's still some problem. :)

**Comment by sideshowbarker** [comment]
@boris-petrov https://repo1.maven.org/maven2/nu/validator/validator/25.12.30/ published

**Comment by boris-petrov** [comment]
All working great now, thanks!
